File: src/main/java/com/zerozero/core/support/error/GlobalErrorType.java
package com.zerozero.core.support.error;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;

@Getter
@RequiredArgsConstructor
public enum GlobalErrorType implements ErrorType {

    INTERNAL_ERROR(HttpStatus.INTERNAL_SERVER_ERROR, "알 수 없는 내부 오류입니다."),
    KAKAO_SERVICE_UNAVAILABLE(HttpStatus.SERVICE_UNAVAILABLE, "카카오 서비스가 응답하지 않습니다."),
    ;

    private final HttpStatus status;

    private final String message;
}

File: src/main/java/com/zerozero/core/support/error/GlobalErrorType.kt
package com.zerozero.core.support.error

import org.springframework.http.HttpStatus

enum class GlobalErrorType(
    override val status: HttpStatus,
    override val message: String,
) : ErrorType {

    INTERNAL_ERROR(HttpStatus.INTERNAL_SERVER_ERROR, "알 수 없는 내부 오류입니다."),
    KAKAO_SERVICE_UNAVAILABLE(HttpStatus.SERVICE_UNAVAILABLE, "카카오 서비스가 응답하지 않습니다."), ;
}
